packages/shared/lib/services/connection.service.ts

```diff
@@ -969,18 +969,20 @@ class ConnectionService {
             instantRefresh ||
             (providerClient.shouldIntrospectToken(providerConfig.provider) && (await providerClient.introspectedTokenExpired(providerConfig, connection)));
 
+        const ExpirationCondition =
+            refreshCondition || (credentials.expires_at && isTokenExpired(credentials.expires_at, template.token_expiration_buffer || 15 * 60));
+
         let tokenExpirationCondition;
 
-        if (template.auth_mode === ProviderAuthModes.OAuth2 || credentials?.type === ProviderAuthModes.OAuth2) {
-            tokenExpirationCondition =
-                credentials.refresh_token &&
-                (refreshCondition || (credentials.expires_at && isTokenExpired(credentials.expires_at, template.token_expiration_buffer || 15 * 60)));
-        } else if (template.auth_mode === ProviderAuthModes.Custom) {
-            tokenExpirationCondition =
-                refreshCondition || (credentials.expires_at && isTokenExpired(credentials.expires_at, template.token_expiration_buffer || 15 * 60));
-        } else if (template.auth_mode === ProviderAuthModes.App || template.auth_mode === ProviderAuthModes.AppStore) {
-            tokenExpirationCondition =
-                refreshCondition || (credentials.expires_at && isTokenExpired(credentials.expires_at, template.token_expiration_buffer || 15 * 60));
+        if ((template.auth_mode === ProviderAuthModes.OAuth2 || credentials?.type === ProviderAuthModes.OAuth2) && providerConfig.provider !== 'facebook') {
+            tokenExpirationCondition = credentials.refresh_token && ExpirationCondition;
+        } else if (
+            (template.auth_mode as ProviderAuthModes) === ProviderAuthModes.Custom ||
+            (template.auth_mode as ProviderAuthModes) === ProviderAuthModes.App ||
+            (template.auth_mode as ProviderAuthModes) === ProviderAuthModes.AppStore ||
+            ((template.auth_mode as ProviderAuthModes) === ProviderAuthModes.OAuth2 && providerConfig.provider === 'facebook')
+        ) {
+            tokenExpirationCondition = ExpirationCondition;
```

Comment:
is the else if actually necessary. Cannot `tokenExpirationCondition` be set to `refreshCondition || (credentials.expires_at && isTokenExpired(...)` by default and set it to `credentials.refresh_token && tokenExpirationCondition` when oauth2 and not facebook